### PR TITLE
Time detection refactor tests

### PR DESCRIPTION
**Pull request by krupalmodi18**
This PR contains changes related to following 
1) Refactoring Time detection code
2) Adding test cases

### CODE REVIEW

File: ner_v2/tests/temporal/time/test_time_detection.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            'hh': int,
+            'mm': int,
+            'range': six.text_type,
+            'time_type': six.text_type,
+            'original_text': six.text_type
+        }
+        df = pd.read_csv(self.csv_path, encoding='utf-8', dtype=dtype, keep_default_na=False)
+
+        for language, language_tests_df in df.groupby(by=['language']):
+            print('Running tests for language {}'.format(language))
```
Comment (srajdev) [inline ner_v2/tests/temporal/time/test_time_detection.py]: E1601 print statement used

File: ner_v2/detectors/temporal/time/standard_time_regex.py [hunk 2]
```diff
@@ -215,7 +256,8 @@ def _detect_hour_minute(self, time_list, original_list):
                 mm = self._get_float_from_numeral(time_match[5])
 
             if time_match[7]:
-                ref_date = self.now_date + int(self.datetime_constant_dict[time_match[7]][1]) * \
+                ref_date = self.now_date + \
+                           int(self.datetime_constant_dict[time_match[7]][1]) * \
```
Comment (srajdev) [inline ner_v2/detectors/temporal/time/standard_time_regex.py]: E126 continuation line over-indented for hanging indent

File: ner_v2/tests/temporal/time/test_time_detection.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            time_detector = TimeDetector(language=language)
+            for index, row in language_tests_df.iterrows():
+                if not self.is_nan(row[TimeDetectorTest.BOT_MESAGE]):
+                    time_detector.set_bot_message(bot_message=row[TimeDetectorTest.BOT_MESAGE])
+                range_enabled = not self.is_nan(row[TimeDetectorTest.RANGE])
+                expected_times, expected_original_texts = self._make_expected_output(row_dict=row)
+                expected_output = list(zip(expected_times, expected_original_texts))
+                detected_times, original_texts = time_detector.detect_entity(text=row[TimeDetectorTest.TEXT],
+                                                                                 range_enabled=range_enabled)
```
Comment (srajdev) [inline ner_v2/tests/temporal/time/test_time_detection.py]: E127 continuation line over-indented for visual indent

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/detectors/temporal/time/standard_time_regex.py [hunk 4]
```diff
@@ -1,18 +1,20 @@
 # coding=utf-8
-from chatbot_ner.config import ner_logger
-from ner_v2.detectors.temporal.constant import DATETIME_CONSTANT_FILE, ADD_DIFF_DATETIME_TYPE, \
-    NUMERALS_CONSTANT_FILE, TIME_CONSTANT_FILE, REF_DATETIME_TYPE, HOUR_TIME_TYPE, MINUTE_TIME_TYPE, \
-    DAYTIME_MERIDIAN, AM_MERIDIEM, PM_MERIDIEM, TWELVE_HOUR
-import pytz
-import re
 import datetime
```
Comment (srajdev) [inline ner_v2/detectors/temporal/time/standard_time_regex.py]: W1618 import missing `from __future__ import absolute_import`

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/tests/temporal/time/test_time_detection.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    def test_time_detection_from_csv(self):
+        """
+        Run time detection tests defined in the csv
+        """
+        # columns = ['language', 'text', 'bot_message', 'hh', 'mm', 'nn', 'range', 'time_type', 'original_text']
+        df = pd.read_csv(self.csv_path, encoding='utf-8', dtype=six.text_type, keep_default_na=False)
+
+        for language, language_tests_df in df.groupby(by=[TimeDetectorTest.LANGUAGE]):
+            print('Running tests for language {}'.format(language))  # noqa: E1601
```
Comment (srajdev) [inline ner_v2/tests/temporal/time/test_time_detection.py]: E1601 print statement used

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Pull request merged**